src/reflect/value.go

```diff
@@ -779,8 +779,17 @@ func (v Value) MapKeys() []Value {
 	k := New(v.typecode.Key())
 	e := New(v.typecode.Elem())
 
+	keyType := v.typecode.Key().(*rawType)
+	isKeyStoredAsInterface := keyType.Kind() != String && !keyType.isBinary()
+
 	for hashmapNext(v.pointer(), it, k.value, e.value) {
-		keys = append(keys, k.Elem())
+		if isKeyStoredAsInterface {
+			intf := *(*interface{})(k.value)
+			v := ValueOf(intf)
+			keys = append(keys, v)
```

Comment:
What about maps that actually do have interface keys, like `map[interface{}]int`? Do they also work with this change?
Might be useful to add a test case for that as well.

Reply:
Added another test for maps with interface keys.